### ISSUE TITLE
RFE: add support for pidfd_open()

### ISSUE THREAD

**Issue by mikkorantalainen** [diazona/pwait#4]
Modern Linux kernels support pidfd_open() which will result in very simple and very stable implementation if available.

For details, see https://man7.org/linux/man-pages/man2/pidfd_open.2.html

**Comment by diazona** [comment]
Good idea, thanks! I did some work on other implementations (e.g. netfilter) and this would fit in with that initiative. When I get a chance I will be happy to work on this, but it might be a while until I have time; alternatively, I'd be quite willing to work with somebody on a pull request.

**Comment by diazona** [comment]
I'm thinking about getting back to this project soon and I will definitely look at this as one of the first features to implement.

**Comment by diazona** [comment]
I finally got around to looking at this, and it seems that `pidfd_open()` does not allow capturing the exit code of the process (unless I'm missing something). I'm still fine with adding it, since I'm also adding a polling mode in #11 and `pidfd_open()` is still going to be better than polling, but it's not nearly as capable as the existing modes netlink or ptrace.
